CMakeLists.txt

```diff
@@ -93,13 +96,26 @@ set(QUDA_INTERFACE_TIFR OFF CACHE BOOL "build tifr interface")
 #QKXTM: DMH
 set(QUDA_INTERFACE_QKXTM ON CACHE BOOL "build QKXTM interface")
 set(QUDA_QKXTM_ARPACK OFF CACHE BOOL "enable QKXTM ARPACK")
+
 if(QUDA_INTERFACE_QKXTM)
-  # Locations for GSL HDF5 LIME OPENBLAS
+  # Locations for GSL HDF5 LIME
   set(QUDA_QKXTM_GSLHOME "" CACHE PATH "path to GSL")
   set(QUDA_QKXTM_LIMEHOME "" CACHE PATH "path to LIME")
   set(QUDA_QKXTM_HDF5HOME "" CACHE PATH "path to HDF5")
-  set(QUDA_QKXTM_OPENBLAS_BUILD "" CACHE PATH "path to OPENBLAS_BUILD")
-  set(QUDA_QKXTM_OPENBLAS_SRC "" CACHE PATH "path to OPENBLAS_SRC")
+
+#-C.K. Locations for OPENBLAS or MKL, depending on GNU/OpenBlas, Intel/MKL switches
+  if(${QUDA_QKXTM_GNU-BUILD})
+    set(QUDA_QKXTM_OPENBLAS_BUILD "" CACHE PATH "path to OPENBLAS_BUILD")
+    set(QUDA_QKXTM_OPENBLAS_SRC "" CACHE PATH "path to OPENBLAS_SRC")
+  endif()
+
+#-C.K. In facts, it seems that these options do not really contribute something,
+# but may be useful for future iterations
+  if(${QUDA_QKXTM_INTEL-BUILD})
+    set(QUDA_QKXTM_MKL_LIBDIR "" CACHE PATH "path to MKL library dir")
+    set(QUDA_QKXTM_MKL_INCDIR "" CACHE PATH "path to MKL include dir")
+  endif()
+#-C.K.
```

Comment:
Exactly what? If one is lucky enough to have a smart enough module loader, these explicit steps are not necessary. Testing and development can be done on smaller machines, whereupon QUDA may need to know explicitly where libraries and headers are located. Magma can be configured without the $MKLROOT environment variable being set. Also, one can have several versions of MKL on a given machine. This gives the user the option to choose versions of MKL. So these options really do contribute something, and will be useful in all future iterations.